service/routes.py

```diff
@@ -58,8 +58,11 @@ def list_customers():
     app.logger.info("Request for customer list")
     customers = []
     email = request.args.get("email")
+    name = request.args.get("name")
     if email:
         customers = Customer.find_by_email(email)
+    elif name:
+        customers = customers.find_by_name(name)
```

Comment:
you will have to define the `find_by_name` function or, rather, `find_by_first_name`  (see above comment), before you can call the method.

to define the `find_by_first_name` function, go into the `models.py` file and create a new function similar to the `find_by_email` function on ln 139.

you can also add the function decorator `@classmethod` above the function (see `find_by_email` for reference). that way, you don't have to use a instance of the `Customer` class, you can just call `Customer.find_by_first_name(first_name)`.  (Note: currently you are calling the `find_by_name` method on a list. while the list is indeed a list of customers, the list class will not inherit the methods defined in the `Customer` class.)